### ISSUE TITLE
Nullable entity comparison with null and implicit/cross joins

### ISSUE THREAD

**Issue by ohnefuenfter** [nhibernate/nhibernate-core#3191]
I have found that the following HQL

```sql
select 1 from CustomEntity08 c where 
 c.Field='ApprovedByBusinessAreaResponsible' and 
 c.ChangeFlow.CustomEntity02PK=94359664 and 
 c.Phase.CustomEntity02PK=null and 
 c.Rule.FieldValue='Mandatory' 
 and c.InUse=1
```

(the mapping is generated by code when session factory is created, i can only tell that the ChangeFlow, Phase and Rule are references to other entities)

will generate the wrong SQL

```sql 
select 1 as col_0_0_ from Custom_Entity08 customenti0_, 
Custom_Entity02 customenti1_ left outer join Custom_Entity02 customenti2_ on customenti0_.Phase=customenti2_.custom_entity02_pk, 
Custom_Entity02 customenti3_ 
where customenti0_.ChangeFlow=customenti1_.custom_entity02_pk 
and customenti0_.Rule=customenti3_.custom_entity02_pk 
and customenti0_.Field='ApprovedByDepartmentChangeManager' 
and customenti1_.custom_entity02_pk=94359664 
and (customenti2_.custom_entity02_pk is null) 
and customenti3_.FieldValue='Mandatory' 
and customenti0_.InUse=1
```

that results in 

```
Msg 4104, Level 16, State 1, Line 1
The multi-part identifier "customenti0_.Phase" could not be bound.
```

Looks like the **left outer join Custom_Entity02 customenti2_** is misplaced?

**Comment by hazzik** [comment]
Please check development version on master branch. There were a lot of improvements in this area.

**Comment by ohnefuenfter** [comment]
We're product company we cannot rely on development versions 
Now we have to rollback to 5.3.9 where there is no problem with the query

**Comment by bahusoid** [comment]
Closing as obsolete. It's fixed in 5.4 and eventually will be fixed in 5.3.15 by #3187
